js-client/src/app-main.ts

```diff
@@ -1,18 +1,16 @@
 import { LitElement, css, html } from "lit";
-import { customElement, query } from "lit/decorators.js";
+import { customElement, queryAsync, state } from "lit/decorators.js";
 import "@shoelace-style/shoelace/dist/themes/light.css";
+import type { ClickItemPayload } from "./nix-search";
 import "./nix-search.ts";
 
-import cytoscape from "cytoscape";
 import dagre from "cytoscape-dagre";
+import cytoscape from "cytoscape";
 
-const API_URL = "/api";
-
-interface GraphEv {
-  name: string;
-}
+const API_URL = __API_URL__;
+cytoscape.use(dagre);
```

Comment:
Can that fail? If so, can leaving that in the global scope lead to unrendered components?

Reply:
You mean the `__API_URL__`? It would be always replaced by a string, with a fallback to `"/api"`:
https://github.com/tweag/nixpkgs-graph-explorer/blob/9847650718357b20bb28987a7b6cc8ca5ff8f70b/js-client/vite.config.ts#L26-L27

Reply:
I meant

```
cytoscape.use(dagre);
```

in React I'd use some sort of context component (see [`useContext`](https://react.dev/reference/react/useContext)) with a [`useEffect`](https://react.dev/reference/react/useEffect) to call this `.use` only when relevant